cacheops/redis.py

```diff
@@ -25,115 +42,74 @@ def handle_connection_failure(call):
             warnings.warn("The cacheops cache timed out! Error: %s" % e, RuntimeWarning)
         except Exception as e:
             warnings.warn("".join(traceback.format_exception(*sys.exc_info())))
-else:
-    handle_connection_failure = identity
 
-client_class_name = getattr(settings, 'CACHEOPS_CLIENT_CLASS', None)
-client_class = import_string(client_class_name) if client_class_name else redis.StrictRedis
-read_clients = []
+    class SafeRedis(client_class):
+        get = handle_connection_failure(client_class.get)
 
+    client_class = SafeRedis
 
-def ip(url):
-    try:
-        return socket.gethostbyname(urlparse(url).hostname)
-    except socket.gaierror:
-        warnings.warn('Hostname in URL %s did not resolve' % url)
-        raise
 
+if settings.CACHEOPS_REDIS_REPLICAS:
```

Comment:
And now we can put all our read-replica code in one place! If there are any replicas in the settings, we subclass the `client_class` to handle them.